### PR TITLE
Fixed some bugs found by the coverity tool

### PR DESCRIPTION
**Pull request by micrakow**
Signed-off-by: Michal Rakowski <michal.rakowski@intel.com>

### CODE REVIEW

File: src/metadata/metadata_raw.c [hunk 1]
```diff
@@ -380,7 +380,7 @@ static int _raw_ram_flush_do_asynch_fill(ocf_cache_t cache,
 	raw = ctx->raw;
 	ENV_BUG_ON(!raw);
 
-	size = raw->entry_size * raw->entries_in_page;
+	size = (uint64_t) raw->entry_size * raw->entries_in_page;
```
Comment (robertbaldyga) [inline src/metadata/metadata_raw.c]: `size` doesn't need to be 64-bit long. In fact total size of entries within page can never be bigger than 4096 bytes, so even `uint16_t` should be more than enough.

File: src/mngt/ocf_mngt_cache.c [hunk 2]
```diff
@@ -1654,14 +1659,8 @@ int ocf_mngt_cache_start(ocf_ctx_t ctx, ocf_cache_t *cache,
...
-	} else {
-		if (cfg->name) {
-			ocf_log(ctx, log_err, "Inserting cache %s failed\n",
-					cfg->name);
-		} else {
-			ocf_log(ctx, log_err, "Inserting cache failed\n");
-		}
-	}
+	} else
+		ocf_log(ctx, log_err, "Inserting cache failed\n");
```
Comment (robertbaldyga) [inline src/mngt/ocf_mngt_cache.c]: As cache name is mandatory now, I'd rather leave the variant with name, which is more informative.

File: src/mngt/ocf_mngt_io_class.c [hunk 3]
```diff
@@ -295,8 +295,8 @@ int ocf_mngt_cache_io_classes_configure(ocf_cache_t cache,
 
 out_edit:
 	if (result) {
-		ENV_BUG_ON(env_memcpy(cache->user_parts, sizeof(&cache->user_parts),
-					old_config, sizeof(&cache->user_parts)));
+		ENV_BUG_ON(env_memcpy(cache->user_parts, sizeof(cache->user_parts),
```
Comment (robertbaldyga) [inline src/mngt/ocf_mngt_io_class.c]: I believe we have similar issue in line 280.

**Comment by micrakow** [comment]
retest this please

**Comment by micrakow** [comment]
retest this please

**Comment by micrakow** [comment]
retest this please

**Pull request merged**